R/certificates.R

```diff
@@ -0,0 +1,94 @@
+
+createCertificateFile <- function(certificate) {
+  certificateFile <- NULL
+
+  # check the R option first, then fall back on the environment variable
+  systemStore <- getOption("rsconnect.ca.bundle")
+  if (is.null(systemStore) || nzchar(systemStore)) {
+    systemStore <- Sys.getenv("RSCONNECT_CA_BUNDLE")
+  }
+
+  # start by checking for a cert file specified in an environment variable
+  if (nzchar(systemStore) && file.exists(systemStore)) {
+    certificateFile <- systemStore
+  }
+
+  # if no certificate contents specified, we're done
+  if (is.null(certificate))
+    return(certificateFile)
+
+  # if we don't have a certificate file yet, try to find the system store
+  if (is.null(certificateFile)) {
+    if (.Platform$OS.type == "unix") {
+      # search known locations on Unix-like
+      stores <- c("/etc/ssl/certs/ca-certificates.crt",
+                  "/etc/pki/tls/certs/ca-bundle.crt",
+                  "/usr/share/ssl/certs/ca-bundle.crt",
+                  "/usr/local/share/certs/ca-root.crt",
+                  "/etc/ssl/cert.pem",
+                  "/var/lib/ca-certificates/ca-bundle.pem")
+    } else {
+      # mirror behavior of curl on Windows, which looks in system folders,
+      # the working directory, and %PATH%
+      stores <- c("C:/Windows/System32/curl-ca-bundle.crt",
+                  "C:/Windows/curl-ca-bundle.crt",
+                  file.path(getwd(), "curl-ca-bundle.crt"),
```

Comment:
I wonder if this should be checked before the default system stores? (So that a curl certificate in the current directory could be used in lieu of the system store.) Or does this reflect pre-existing behavior for curl on Windows already?

Reply:
Just checked; it looks like curl on Windows does check the current working directory first (and otherwise follows the precedence here). I'll use that permutation.